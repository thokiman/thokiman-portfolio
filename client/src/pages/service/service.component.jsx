import React, { lazy, Suspense } from "react";
import { Switch, Route } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import PageShell from "../../components/page-shell/page-shell.component";
import { ServiceContainer, ServiceSideBarHide } from "./service.styles";
import ServiceGraphicDesign from "../../components/service-graphic-design/service-graphic-design.component";
import {
  selectArtService,
  selectEngineerService,
  selectFullstackService,
  selectPhotoService,
  selectServiceRoute,
} from "../../redux/service/service.selectors";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors.js";
import SpinnerLoading from "../../components/spinner-loading/spinner-loading.component";
const ServiceHomeContent = lazy(() =>
  import("../../components/service-content/service-content.component")
);
const ServiceHeader = lazy(() =>
  import("../../components/service-header/service-header.component")
);
const ServiceEngineer = lazy(() =>
  import("../../components/service-engineer/service-engineer.component")
);
const ServicePhotography = lazy(() =>
  import("../../components/service-photography/service-photography.component")
);
const ServiceFullStackDeveloper = lazy(() =>
  import(
    "../../components/service-full-stack-developer/service-full-stack-developer.component"
  )
);

export const Service = ({
  art,
  photo,
  engineer,
  fullstack,
  serviceRoute,
  isSideBarHidden,
}) => {
  return (
    <ServiceContainer $issidebarhidden={isSideBarHidden}>
      {isSideBarHidden ? (
        <ServiceSideBarHide>
          Serv
          <br />
          ice
        </ServiceSideBarHide>
      ) : null}
      <ServiceHeader />
      <Switch>
        <Suspense fallback={<SpinnerLoading />}>
          <Route exact path={serviceRoute} component={ServiceHomeContent} />
          <Route path={engineer.routeName} component={ServiceEngineer} />
          <Route
            path={fullstack.routeName}
            component={ServiceFullStackDeveloper}
          />
          <Route path={art.routeName} component={ServiceGraphicDesign} />
          <Route path={photo.routeName} component={ServicePhotography} />
        </Suspense>
      </Switch>
    </ServiceContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  art: selectArtService,
  photo: selectPhotoService,
  engineer: selectEngineerService,
  fullstack: selectFullstackService,
  serviceRoute: selectServiceRoute,
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(PageShell(Service));
