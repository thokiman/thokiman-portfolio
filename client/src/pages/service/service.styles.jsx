import styled, { css, keyframes } from 'styled-components';

const getServiceImage = ({ $bigWaveImage }) => {
  return $bigWaveImage.item.iconPath;
};

const serviceSideBarHide = keyframes`
  0% {
    transform: rotateX(100deg);
    transform-origin: bottom;
    opacity: 0;
  }
  100% {
    transform: rotateX(0);
    transform-origin: bottom;
    opacity: 1;
  }

`;

export const ServiceSideBarHide = styled.div`
  font-family: futura-demi;
  color: #e3e1e4;
  position: absolute;
  top: 10vh;
  left: 6vw;
  font-size: 480%;
  letter-spacing: 0.17em;
  width: 50vw;
  line-height: 1.5em;
  animation: ${serviceSideBarHide} 2s cubic-bezier(0.175, 0.885, 0.32, 1.275)
    both;
`;
const ServiceSideBarActive = css`
  @media only screen and (max-width: 800px) {
    position: relative;
    left: 0;
    transition: all 0.5s ease;
    opacity: 1;
    width: 100vw;
    height: 100vh;
    flex-direction: column;
  }
`;
const getServiceSideBarActive = (props) => {
  if (props.$issidebarhidden) return null;
  return ServiceSideBarActive;
};

export const ServiceContainer = styled.div`
  background-image: url(${getServiceImage});
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 72vh;
  transition: background-image 0.7s ease-in-out, height 0.4s ease-in-out;
  font-family: futura-book;
  opacity: 0.9;
  @media only screen and (max-width: 800px) {
    width: 64vw;
    position: relative;
    top: 0;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: all 0.5s ease;
    height: 115vh;
    overflow-y: scroll;
    overflow-x: hidden;
    opacity: 0.5;
    height: 70.8vh;
  }
  ${getServiceSideBarActive};
`;

ServiceContainer.displayName = 'ServiceContainer';
