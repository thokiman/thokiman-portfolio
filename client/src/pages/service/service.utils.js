import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize4,
  hasScreenSize5,
} from '../../utils/common.component.utils';

export const measureServiceHeight = (
  pathname,
  viewWidth,
  viewHeight,
  serviceRoute,
  engineerRoute,
  artRoute,
  devRoute,
  photoRoute
) => {
  let height;
  console.log('--', viewWidth);
  console.log(viewHeight);
  console.log('**', pathname);
  console.log('##', serviceRoute);
  console.log('##', engineerRoute);
  console.log('##', artRoute);
  console.log('##', devRoute);
  console.log('##', photoRoute);
  switch (pathname) {
    case serviceRoute:
      if (hasScreenSize1(viewWidth)) {
        height = 140;
      } else if (hasScreenSize2(viewWidth)) {
        height = 140;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        height = 140;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        height = 150;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        height = 140;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        height = 140;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        height = 140;
      } else if (hasScreenSize5(viewWidth)) {
        height = 140;
      } else {
        height = 140;
      }
      return height;

    case engineerRoute:
      if (hasScreenSize1(viewWidth)) {
        height = 295;
      } else if (hasScreenSize2(viewWidth)) {
        height = 295;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        height = 225;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        height = 223;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        height = 255;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        height = 255;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        height = 269;
      } else if (hasScreenSize5(viewWidth)) {
        height = 253;
      } else {
        height = 255;
      }

      return height;
    case artRoute:
      if (hasScreenSize1(viewWidth)) {
        height = 265;
      } else if (hasScreenSize2(viewWidth)) {
        height = 250;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        height = 195;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        height = 195;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        height = 210;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        height = 220;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        height = 230;
      } else if (hasScreenSize5(viewWidth)) {
        height = 215;
      } else {
        height = 222.5;
      }
      return height;
    case devRoute:
      if (hasScreenSize1(viewWidth)) {
        height = 400;
      } else if (hasScreenSize2(viewWidth)) {
        height = 373;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        height = 263;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        height = 260;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        height = 315;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        height = 318;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        height = 318;
      } else if (hasScreenSize5(viewWidth)) {
        height = 353;
      } else {
        height = 309.4;
      }
      return height;
    case photoRoute:
      if (hasScreenSize1(viewWidth)) {
        height = 370;
      } else if (hasScreenSize2(viewWidth)) {
        height = 360;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        height = 265;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        height = 260;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        height = 305;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        height = 305;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        height = 320;
      } else if (hasScreenSize5(viewWidth)) {
        height = 310;
      } else {
        height = 311.9;
      }
      return height;

    default:
      return 800;
  }
};
