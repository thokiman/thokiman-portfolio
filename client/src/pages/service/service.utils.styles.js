import {
  hasScreenSize1,
  hasScreenSize1_1,
  hasScreenSize1_2,
  hasScreenSize1_3,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
  hasScreenSizeWindow_1,
} from '../../utils/common.component.utils';

export const measureServiceIsNotActivated = (viewWidth, viewHeight) => {
  let serviceIsNotActivatedProps;

  if (hasScreenSize1(viewWidth)) {
    serviceIsNotActivatedProps = {
      top: 30,
      left: 6,
      fontSize: 7,
    };
  } else if (hasScreenSize2(viewWidth)) {
    serviceIsNotActivatedProps = {
      top: 41,
      left: 6,
      fontSize: 5,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    serviceIsNotActivatedProps = {
      top: 36.5,
      left: 6,
      fontSize: 8,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    serviceIsNotActivatedProps = {
      top: 37,
      left: 6,
      fontSize: 9,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    serviceIsNotActivatedProps = {
      top: 40,
      left: 6,
      fontSize: 7,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    serviceIsNotActivatedProps = {
      top: 44,
      left: 6,
      fontSize: 6,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    serviceIsNotActivatedProps = {
      top: 50,
      left: 6,
      fontSize: 5,
    };
  } else if (hasScreenSize5(viewWidth)) {
    serviceIsNotActivatedProps = {
      top: 43.5,
      left: 6,
      fontSize: 12,
    };
  } else {
    serviceIsNotActivatedProps = {
      top: 36,
      left: 6,
      fontSize: 10,
    };
  }
  return serviceIsNotActivatedProps;
};

export const measureServiceHeight = (
  pathname,
  viewWidth,
  viewHeight,
  serviceRoute,
  engineerRoute,
  artRoute,
  devRoute,
  photoRoute
) => {
  let height;
  const serviceOffset = 100;
  const engineerOffset = 50;
  const artOffset = 50;
  const devOffset = 50;
  const photoOffset = 50;

  switch (pathname) {
    case serviceRoute:
      if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
        console.log('service route', 1);
        height = 140 + serviceOffset;
      } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
        console.log('service route', 2);
        height = 140 + serviceOffset;
      } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
        console.log('service route', 3);
        height = 140 + serviceOffset;
      } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
        console.log('service route', 4);
        height = 140 + serviceOffset;
      } else if (hasScreenSize1(viewWidth)) {
        console.log('service route', 5);
        height = 140 + serviceOffset;
      } else if (hasScreenSize2(viewWidth)) {
        console.log('service route', 6);
        height = 140 + serviceOffset;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        console.log('service route', 7);
        height = 140 + serviceOffset;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        console.log('service route', 8);
        height = 100 + serviceOffset;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        console.log('service route', 9);
        height = 120 + serviceOffset;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        console.log('service route', 10);
        height = 140 + serviceOffset;
      } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
        console.log('service route', 11);
        height = 140 + serviceOffset;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        console.log('service route', 12);
        height = 160 + serviceOffset;
      } else if (hasScreenSize5(viewWidth)) {
        console.log('service route', 13);
        height = 80 + serviceOffset;
      } else {
        console.log('service route', 14);
        height = 300;
      }
      return height;

    case engineerRoute:
      if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
        console.log('engineer route', 1);
        height = 295 + engineerOffset;
      } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
        console.log('engineer route', 2);
        height = 295 + engineerOffset;
      } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
        console.log('engineer route', 3);
        height = 295 + engineerOffset;
      } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
        console.log('engineer route', 4);
        height = 295 + engineerOffset;
      } else if (hasScreenSize1(viewWidth)) {
        console.log('engineer route', 5);
        height = 340 + engineerOffset;
      } else if (hasScreenSize2(viewWidth)) {
        console.log('engineer route', 6);
        height = 320 + engineerOffset;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        console.log('engineer route', 7);
        height = 240 + engineerOffset;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        console.log('engineer route', 8);
        height = 223 + engineerOffset;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        console.log('engineer route', 9);
        height = 270 + engineerOffset;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        console.log('engineer route', 10);
        height = 280 + engineerOffset;
      } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
        console.log('engineer route', 11);
        height = 255 + engineerOffset;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        console.log('engineer route', 12);
        height = 290 + engineerOffset;
      } else if (hasScreenSize5(viewWidth)) {
        console.log('engineer route', 13);
        height = 280 + engineerOffset;
      } else {
        console.log('engineer route', 14);
        height = 450;
      }

      return height;
    case artRoute:
      if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
        console.log('art route', 1);
        height = 265 + artOffset;
      } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
        console.log('art route', 2);
        height = 265 + artOffset;
      } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
        console.log('art route', 3);
        height = 265 + artOffset;
      } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
        console.log('art route', 4);
        height = 265 + artOffset;
      } else if (hasScreenSize1(viewWidth)) {
        console.log('art route', 5);
        height = 310 + artOffset;
      } else if (hasScreenSize2(viewWidth)) {
        console.log('art route', 6);
        height = 280 + artOffset;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        console.log('art route', 7);
        height = 195 + artOffset;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        console.log('art route', 8);
        height = 195 + artOffset;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        console.log('art route', 9);
        height = 225 + artOffset;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        console.log('art route', 10);
        height = 240 + artOffset;
      } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
        console.log('art route', 11);
        height = 220 + artOffset;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        console.log('art route', 12);
        height = 250 + artOffset;
      } else if (hasScreenSize5(viewWidth)) {
        console.log('art route', 13);
        height = 215 + artOffset;
      } else {
        console.log('art route', 14);
        height = 410; // need adjust
      }
      return height;
    case devRoute:
      if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
        console.log('dev route', 1);
        height = 400 + devOffset;
      } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
        console.log('dev route', 2);
        height = 400 + devOffset;
      } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
        console.log('dev route', 3);
        height = 400 + devOffset;
      } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
        console.log('dev route', 4);
        height = 400 + devOffset;
      } else if (hasScreenSize1(viewWidth)) {
        console.log('dev route', 5);
        height = 450 + devOffset;
      } else if (hasScreenSize2(viewWidth)) {
        console.log('dev route', 6);
        height = 400 + devOffset;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        console.log('dev route', 7);
        height = 300 + devOffset;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        console.log('dev route', 8);
        height = 250 + devOffset;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        console.log('dev route', 9);
        height = 330 + devOffset;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        console.log('dev route', 10);
        height = 330 + devOffset;
      } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
        console.log('dev route', 11);
        height = 318 + devOffset;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        console.log('dev route', 12);
        height = 330 + devOffset;
      } else if (hasScreenSize5(viewWidth)) {
        console.log('dev route', 13);
        height = 350 + devOffset;
      } else {
        console.log('dev route', 14);
        height = 500;
      }
      return height;
    case photoRoute:
      if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
        console.log(' photoroute', 1);
        height = 370 + photoOffset;
      } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
        console.log(' photoroute', 2);
        height = 370 + photoOffset;
      } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
        console.log(' photoroute', 3);
        height = 370 + photoOffset;
      } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
        console.log(' photoroute', 4);
        height = 370 + photoOffset;
      } else if (hasScreenSize1(viewWidth)) {
        console.log(' photoroute', 5);
        height = 410 + photoOffset;
      } else if (hasScreenSize2(viewWidth)) {
        console.log(' photoroute', 6);
        height = 400 + photoOffset;
      } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
        console.log(' photoroute', 7);
        height = 265 + photoOffset;
      } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
        console.log(' photoroute', 8);
        height = 260 + photoOffset;
      } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
        console.log(' photoroute', 9);
        height = 370;
      } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
        console.log(' photoroute', 10);
        height = 330 + photoOffset;
      } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
        console.log(' photoroute', 11);
        height = 305 + photoOffset;
      } else if (hasScreenSize4(viewWidth, viewHeight)) {
        console.log(' photoroute', 12);
        height = 335 + photoOffset;
      } else if (hasScreenSize5(viewWidth)) {
        console.log(' photoroute', 13);
        height = 310 + photoOffset;
      } else {
        console.log(' photo route', 14);
        height = 535;
      }
      return height;

    default:
      return 800;
  }
};
