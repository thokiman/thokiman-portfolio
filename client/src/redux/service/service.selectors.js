import { createSelector } from 'reselect';

const selectAllService = (state) => state.service.service;
const selectAllOther = (state) => state.service.other;

export const selectArtService = createSelector(
  [selectAllService],
  (service) => service.art
);
export const selectPhotoService = createSelector(
  [selectAllService],
  (service) => service.photo
);
export const selectEngineerService = createSelector(
  [selectAllService],
  (service) => service.engineer
);
export const selectDevService = createSelector(
  [selectAllService],
  (service) => service.fullstack
);

export const selectServiceRoute = createSelector(
  [selectAllService],
  (service) => service.routeName
);
export const selectEngineerServiceRoute = createSelector(
  [selectAllService],
  (service) => service.engineer.routeName
);

export const selectArtServiceRoute = createSelector(
  [selectAllService],
  (service) => service.art.routeName
);

export const selectPhotoServiceRoute = createSelector(
  [selectAllService],
  (service) => service.photo.routeName
);

export const selectDevServiceRoute = createSelector(
  [selectAllService],
  (service) => service.fullstack.routeName
);

export const selectBigWaveImage = createSelector(
  [selectAllOther],
  (other) => other.layout.items.bigWave
);
