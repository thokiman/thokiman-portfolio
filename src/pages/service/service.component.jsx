import React from "react";
import { Switch, Route } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import PageShell from "../../components/page-shell/page-shell.component";
import ServiceHomeContent from "../../components/service-content/service-content.component";
import ServiceHeader from "../../components/service-header/service-header.component";
import ServiceEngineer from "../../components/service-engineer/service-engineer.component";
import ServicePhotography from "../../components/service-photography/service-photography.component";
import ServiceFullStackDeveloper from "../../components/service-full-stack-developer/service-full-stack-developer.component";
import "./service.styles.scss";
import ServiceGraphicDesign from "../../components/service-graphic-design/service-graphic-design.component";
import {
  selectArtService,
  selectEngineerService,
  selectFullstackService,
  selectPhotoService,
  selectServiceRoute,
} from "../../redux/service/service.selectors";

const Service = ({ art, photo, engineer, fullstack, serviceRoute }) => {
  return (
    <div className="service-container">
      <ServiceHeader />
      <Switch>
        <Route
          exact
          path={serviceRoute}
          render={() => <ServiceHomeContent />}
        />
        <Route
          path={engineer.routeName}
          render={() => <ServiceEngineer engineer={engineer} />}
        />
        <Route
          path={fullstack.routeName}
          render={() => <ServiceFullStackDeveloper fullstack={fullstack} />}
        />
        <Route
          path={art.routeName}
          render={() => <ServiceGraphicDesign art={art} />}
        />
        <Route
          path={photo.routeName}
          render={() => <ServicePhotography photo={photo} />}
        />
      </Switch>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  art: selectArtService,
  photo: selectPhotoService,
  engineer: selectEngineerService,
  fullstack: selectFullstackService,
  serviceRoute: selectServiceRoute,
});

export default connect(mapStateToProps)(PageShell(Service));
