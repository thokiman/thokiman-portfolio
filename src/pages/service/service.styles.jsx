import styled, { css, keyframes } from "styled-components";
import ServiceImage from "../../assets/layout/water-3.jpg";
const serviceSideBarHide = keyframes`
  0% {
    transform: rotateX(100deg);
    transform-origin: bottom;
    opacity: 0;
  }
  100% {
    transform: rotateX(0);
    transform-origin: bottom;
    opacity: 1;
  }

`;

export const ServiceSideBarHide = styled.div`
  font-family: futura-demi;
  color: #e3e1e4;
  position: absolute;
  top: 10vh;
  left: 6vw;
  font-size: 550%;
  letter-spacing: 0.17em;
  width: 50vw;
  line-height: 1.5em;
  animation: ${serviceSideBarHide} 2s cubic-bezier(0.175, 0.885, 0.32, 1.275)
    both;
`;
const ServiceSideBarActive = css`
  @media only screen and (max-width: 800px) {
    position: relative;
    left: 0;
    transition: all 0.5s ease;
    opacity: 1;
    width: 100vw;

    flex-direction: column;
  }
`;
const getServiceSideBarActive = (props) => {
  if (props.$issidebarhidden) return null;
  return ServiceSideBarActive;
};

export const ServiceContainer = styled.div`
  background-image: url(${ServiceImage});
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 72vh;
  transition: background-image 0.7s ease-in-out, height 0.4s ease-in-out;
  font-family: futura-book;
  opacity: 0.9;
  @media only screen and (max-width: 800px) {
    width: 65vw;
    position: relative;
    top: 0;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: all 0.5s ease;
    height: 115vh;
    overflow-y: scroll;
    overflow-x: hidden;
    opacity: 0.5;
  }
  ${getServiceSideBarActive};
`;

ServiceContainer.displayName = "ServiceContainer";
