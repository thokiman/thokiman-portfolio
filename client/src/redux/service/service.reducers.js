import { other } from "../../data.object/other.data";
import { service } from "../../data.object/service.data";

const INITIAL_STATE = {
  service,
  other
};

const serviceReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export default serviceReducer;
