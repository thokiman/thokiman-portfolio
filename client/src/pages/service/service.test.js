import React from "react";
import { shallow } from "enzyme";

import { Service } from "./service.component";

describe("Service page", () => {
  let wrapper;
  const mockRoute = "/";
  const mockObject = [];
  const mockFalseBool = false;
  beforeEach(() => {
    const mockProps = {
      art: mockObject,
      photo: mockObject,
      engineer: mockObject,
      fullstack: mockObject,
      serviceRoute: mockRoute,
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<Service {...mockProps} />);
  });
  it("should render Service page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render ServiceSideBarHide if isSideBarHidden is false", () => {
    expect(wrapper.exists("ServiceSideBarHide")).toBe(false);
  });
});
