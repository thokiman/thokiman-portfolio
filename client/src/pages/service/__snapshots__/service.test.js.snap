// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service page should render Service page 1`] = `ShallowWrapper {}`;
